### PR TITLE
Plugin Version decoupling for MDS support

### PR DESCRIPTION
**Pull request by riysaxen-amzn**
### Description
This PR filters out the versions which are not supported when MDS support is enabled
 
### Issues Resolved
* The Cluster Metrics Monitor part in MDS doesn't work due to Version Compatibility issue.
Cluster Metrics Monitor creation encounters issues with other data sources <=2.11. This discrepancy arises from a version incompatibility. Specifically, a field was introduced in version 2.13 as part of the request payload of _monitors/_execute  API is expected to be absent in version 2.11 data source request payloads. Therefore, we get parsing exception for _monitors/_execute  API from Alerting Backend when we are connected with dataSource versions <= 2.11. Therefore the support range for MDS is >=2.13 for alerting Plugin
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting-dashboards-plugin/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: opensearch_dashboards.json [hunk 1]
```diff
@@ -16,5 +16,7 @@
     "opensearchDashboardsUtils"
   ],
   "server": true,
-  "ui": true
+  "ui": true,
+  "supportedOSDataSourceVersions": ">=2.13.0",
```
Comment (AWSHurneyt) [inline opensearch_dashboards.json:20]: Saw this comment in this other PR https://github.com/opensearch-project/dashboards-notifications/pull/221#discussion_r1680328882. Should this version be adjusted?

Comment (riysaxen-amzn) [reply opensearch_dashboards.json:20]: i have updated the description, why we need >=2.13

**Review by AWSHurneyt — APPROVED** [review]
Approved assuming CI passes.

**Review by ZilongX — APPROVED** [review]

**Comment by riysaxen-amzn** [comment]
that's a flaky test which is failing

**Review by amsiglan — APPROVED** [review]

**Pull request merged**